Dart/src/com/jetbrains/lang/dart/analyzer/DartAnalysisServerService.java

```diff
@@ -225,9 +230,43 @@ public void run(@NotNull ProgressIndicator indicator) {
     @Override
     public void beforeDocumentChange(DocumentEvent e) {
       myServerData.onDocumentChanged(e);
+      synchronized (myLock) {
+        final Document document = e.getDocument();
+        myUpdatedDocuments.add(document);
+        scheduleUpdateFilesContent();
+      }
     }
   };
 
+  private void scheduleUpdateFilesContent() {
```

Comment:
Thinking a little more about it - in general DAS is not limited to analysis just Dart and HTML.

Right now - yes.

I don't know yet all the details, but we might have SASS files, defining names referenced in Angular HTML templates. So, ideally we need to notify DAS about any file in any analysis root managed by DAS.